### DISCUSSION TITLE
Recommended pattern for job-scoped variables?

### DISCUSSION THREAD

**Discussion by gcleaves** [windmill-labs/windmill#2872] [Questions] [answered]
Hi. I'd like to set a variable at the beginning of my flow and reuse it throughout the flow. What is the recommended pattern?

I don't think I can use a standard or state variable because another job could change the variable during parallel executions. I thought about creating a variable at the start of the flow including the contextual JOB_ID, but this will create many variables which I don't know how to programmatically destroy. 

Should I just update the `results` object? `results.my_variable = value` and refer to that throughout?

**Comment by rubenfiszel** [comment, marked as answer]
You cannot modify the results object, it's a proxy object that actually call the api underneath. However you have the right intuition, set it as the result of your first step (by returning it) to achieve the desired effect

**Reply by gcleaves** [reply]
Thanks. Although in practice my "variable" might be set in different steps of the branching flow. So it could be results.d.my_variable or results.m.my_variable. This makes it hard to refer to consistently further down the flow, because I don't know the exact location inside results.

To get around this, I find myself needing to add a step at the end of my branches to set the variable, which feels unnecessary if ephemeral variables could be set.

**Reply by rubenfiszel** [reply]
Keep in mind that the result of a branchone is the result of the branch chosen so you could be consistent about it and have a stable identifier
